.github/workflows/test.yml

```diff
@@ -0,0 +1,39 @@
+name: Verify and Test
+on: 
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+jobs:
+  golangci:
+    strategy:
+      matrix: 
+        go: ['1.18', '1.19','1.20']
+        os: [ubuntu-latest, macos-latest, windows-latest]
+      fail-fast: true
+    name: verify
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      - name: Setup Go ${{ matrix.go }}
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ matrix.go }}
+          cache: false
+      
+      - name: Golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with: 
+          version: v1.53
+          args: --timeout=5m
+      
+      - name: Test
+        run: go test -race --coverprofile=coverage.coverprofile --covermode=atomic -v ./...
```

Comment:
We `might` be able to update this to be `make test`, that way the CI testing stays in sync with the `make test` command that contributors will run locally during development.